### DISCUSSION TITLE
Oxygen

### DISCUSSION THREAD

**Discussion by soerenthomsen** [GliderToolsCommunity/GliderTools#74] [Ideas]
Dear GliderTools Community, 

One big wish for this package is an improved oxygen processing and quality correction etc. 
I am willing to contribute to this as I have some expertise on oxygen due to my PhD work. 

FYI:  @patricialg, @tomhull and myself (@soerenthomsen) and others are drafting a Standard Operational Practice (SOP) document for OceanGliders.org at the moment. 

To make it as useful as possible: 
1) Dear users (i.e. SouthAfrica ( @sarahnicholson, @isgiddy, @tjryankeogh), Sweden Gothenburg GU (@bastienqueste, @sebswart, @marcelduplessis), France, Marseille MIO (@AnthonyBosse), Peru, Lime Imarpe (@AlicePietri) I would like to ask you which platform are you using and which oxygen sensors are attached to these platforms. This helps us to know on what we should focus on first. 
2) People with or without experience on oxygen are welcome to join this effort. 

If you want to support the new oxygen package please go directly here: #112 

💚🌊🌏

**Comment by AlicePietri** [comment]
Thank you Sören for starting this conversation, better tools for processing and correcting oxygen would indeed be extremely helpful. 
For my part I have only used slocum gliders, all equipped with aanderaa optodes.

**Reply by soerenthomsen** [reply]
@AlicePietri could you provide a detailed list of the Aanderaa optodes at IMARPE? @patricialg

**Reply by soerenthomsen** [reply]
@AlicePietri which model do you use?

**Reply by AlicePietri** [reply]
@soerenthomsen, @patricialg we've been flying a Slocum G2 with an Aanderaa 3835. For the G3s I will have to check, we haven't used them yet though.

**Comment by bastienqueste** [comment]
I've worked with the following combos:
Seaglider / Aanderaa 4330
Seaglider / Aandera 4831
Seaglider / Contros Hydroflash
SeaExplorer / Rinko AROD-FT
SeaExplorer / RBR Coda

One thing that will be necessary for the toolbox will be accounting for the fact that many glider integrations don't return raw phase data for optodes and many novice oxygen users don't realise they're measurements of pp(O2) rather than concentrations - and that the concentrations they get are based on an incorrect, fixed and often unknown salinity.

**Reply by richardsc** [reply]
We (DFO Canada) also have SeaExplorer but with the SBE43. I have to say that we've had a lot of issues with it and I think purchasing new we would go with the Rinko/Coda.

@bastienqueste raises some good points though -- the different sensors all record different things and we found (e.g. for the Rinko) even the company reps were not totally clear on what the default output is based on (salinity=35 or salinity=0?). 

I'm interested in your experience with the Coda -- from what I've been told it is *calibrated* for concentration, not pp(O2). And I know when it's integrated with a regular CTD it automatically uses the CTD-measured salinity, but this can be a problem for example if the CTD has an offset or error that needs to be corrected in post-processing.

**Reply by bastienqueste** [reply]
Welcome! Glad to see you here - was going to post something in the SEGUE slack.
We should be getting our first batch of coda data in the fall, until then it's all Rinko. For the Rinko, I believe (haven't had the chance to test as gliders are in the water) that you can communicate directly with the sensor through the SX to double check the salinity setting.

Coda oxygen will require some efforts to qualify quality/calibration.

B

**Comment by bastienqueste** [comment]
I'm happy to contribute to this effort, though won't have the bandwidth for it until May.
Soeren, do I understand correctly that you are planning to make the corrections platform specific? Although I can see some short term benefit (ie. there's a lot of work existing to help the fact that optodes are on the butt of slocums), I think it might limit the toolset long term depending on how it's implemented. What were you thinking there?

Almost all (except 1 I think?) commercially available oxygen sensors that are sold for gliders are optodes working on the same principle and the same equations. Not sure we need to be Aanderaa specific either at the start :)

**Reply by soerenthomsen** [reply]
At the moment I just collect all info to brainstorm what makes most sense before starting to code anything ;) thanks for sharing your thoughts! Yes makes all sense! Tused tack!

**Comment by AnthonyBosse** [comment]
Hi all! I am not an expert on Oxygen but want to learn more. In terms of platform/sensor, french gliders have been deployed with the following configurations :
- Slocum with Anderaa 5013 and 4831
- Seaglider with SBE43F
- Seaexplorer with SBE43F

**Comment by richardsc** [comment]
I am no expert, but also interested in being involved/helping out with this. DFO Canada has:

* SeaExplorer with Rinko
* SeaExplorer with SBE43F

One problem we have had with the real-time data is that the SX+SBE43 outputs only frequency, so we have to code in the calibration coefficients and equations manually to be able to convert to concentration.

A second issue we've had with the SBE43 is that we find SeaBird doesn't properly enter the coefficients on the sensor after we send them for calibration -- we have to connect to it directly to verify and correct (we've had correspondence with them recently that they're going to do better about this ...).

A third issue is that every time we send the GP-CTD+43F back for calibration they are not able to do a pre-cal check because they membrane has broken. Every time. So we actually don't know how good a lot of our data is, given that we didn't regularly collect samples to do Winkler titrations to be able to check/calibrate.

**Comment by patricialg** [comment]
Hi all,
This is a non-coding question  
I was wondering if any of you have any experience with optodes mounted in the CTD rosette (or know someone who can help). 
This is mainly for field calibration when the glider is deployed from a research vessel equipped with a CTD and a calibrated O2 sensor.  The glider optode can either be connected directly to the CTD profiler if able to receive the digital (RS232) output from the Optode (Ushida et al., 2010), or attached via a data logger. Record down- and upcast data to allow hysteresis correction. The Optode data obtained during the bottle-firing stop for collection of water samples can be used for in situ calibration, since the difference between the downcast and upcast oxygen profiles is relatively small ( 1 umol kg-1 approx., Ushida et al., 2008). The error in the Optode can be reduced by allowing sufficient time for the sensor equilibration after the stop (at least 2 min, ref maybe in Hahn et al. 2014??). 
I followed Ushida GO-SHIP recommendations, but it seems that people are using a higher equilibration time,  which I haven’t been able to find in any recent publication. Therefore, my questions are:
-	Do you have any equilibration time you recon is better than that proposed for Ushida 2010?
-	The recommendation is to use CTD/O2 sensor from the rosette to calibrate optode sensor/s and Winkler samples to calibrate the rosette sensor (Hahn et al., 2014). Is that what you do? Or you use the winkler samples for optode calibration also?
-	I assume the bottle firing and optode equilibration time is happening during the upcast. Some people asked me why not in the downcast, I assume you record all values (like when working with the CTD rosette) but not stopping during downcast. Is that correct or you have any experience doing optode equilibration during downcast and upcast?
-	We all know we should calibrate the glider sensor before deployment and after recovery. After recovery includes 0% and 100% calibration (and if possible at different temp). I’ve found some people mention only 0% calibration. Do you have any preference?
Sorry for asking so many questions :P 
Thank you very much for your help!
Hugs,
Patry

**Reply by patricialg** [reply]
I moved here Tom Hull's reply:
Hey, so I've used several on several rosette systems. Some versions of
seabird can take the RS232 directly, but those that can't can still use
4831 series optodes as these have a built in analog out. The seabird
software doesn't support it, so you have to do all the processing with your
own scripts of course, but it's another option.

I tend to do at least a minute on-top of the surface soak and then a minute
at the bottom. In practice looking at some recent profiles its typically
more like two minutes.
But you do raise another important point: the equilibration time is quite
dependent on if the optode dries out between dips, in honesty it can take
several hours for the foil to behave properly if it has dried out
completely!

I'm not sure I understand the upcast-downcast question

0 & 100% is better to set the span (gain), optodes drift with a reduction
in sensitivity in addition to an increasingly positive offset from zero, so
you really need both points to correct it properly.

**Reply by dhruvbalwada** [reply]
thanks for moving @tomhull 's reply here. 
Apologies for policing. :)

**Reply by patricialg** [reply]
Tom, thanks for your reply! Very useful :)

About the sensor drying out, I think it's very important to keep the sensor's membrane wet all time. If sensor is mounted in the glider, then keeping a wet sponge will help. I think if the sensor dries out then I won't trust values for the first 8h (some are faster than others but you never know... they can be quite unstable)

Upcast-downcast: I haven't been involved in a glider deployment from a big ship, so this is only info I've gathered from articles/ people. When you read Ushida (Go-SHIP 2010) and Hahn et al., 2014, they mentioned they mounted the optode in the rosette frame and they let the optode sensor to equilibrate during bottle firing. So, from this info, I understand that you are doing profiles like always, a downcast without stopping (is the speed the same or do you need to reduce speed?) and when you are doing upcast, you stop to close the nisking bottles (in this case, you have to wait 2mins? to let the optode equilibrate before closing the bottle and continue your trip up). I just remember Bittig 2014 (https://aslopubs.onlinelibrary.wiley.com/doi/epdf/10.4319/lom.2014.12.617) and in figure1 you can see that 2 mins is the more conservative response time, maybe this is where 2 mins come from? 

So you also recomend doing 0% and 100% after recovery (and keep sensor membarne wet all time)

Thanks again!! :)

**Reply by patricialg** [reply]
@dhruvbalwada no worries, it makes more sense to have this conversation here :)

**Reply by gkrahmann** [reply]
Hi all.
Gerd Krahmann, from the same group as Johannes Hahn here. We are mounting our Aanderaa optodes to the CTD via a self-registering and self powered logger which we developed in-house for mooring operations. All glider (and mooring) optodes are if possible run through calibration casts before and after deployment and Johannes has developed routines to calculate (replacement) calibration coefficients from a small number of calibration stops. I suspect he uses calibrated CTD oxygen and not Winkler oxygen (If you have one Winkler outlier in 5 stops that would be problematic). 0% oxygen at two temperatures from lab measurements also goes into the calibration.  The calibration coefficients are in difference to Aanderaa's only valid for a limited range! Usually the conditions during the two calibration casts and the two 0% values cover enough of the condition range the optode will encounter during the deployment. So you can not use them for the next deployment unless it is in the same location with the same conditions.
As far as I remember 100% measurements in the lab on the ships were not too reliable, so he dropped them. 
Stops might even be longer than 2 minutes. But only a small number (5 to 10 or so).
The whole system is not the best one can do, but for us a good compromise between ship time and data quality.
For our Slocum gliders with optodes I optimize the required delays on temperature (based on CTD temperature, not optode temperature) so that the difference between subsequent up and down casts are minimal. Usually this results in values between 25 and 35 seconds for slow foils. Optode oxygen is also reverse filtered in time to remove slow sensor response and then refiltered with a forward-backward filter of similar strength (filtfilt in Matlab) to remove the noise.

In the tropics we found that bio-fouling is a big issue. After 2-3 weeks of shallow gliding algae start to grow on the foil and change the oxygen levels during light time (shallow and daylight). Do you have any ideas about that ?

**Comment by patricialg** [comment]
![Capture](https://user-images.githubusercontent.com/13015381/113219387-330b2a80-9279-11eb-8382-4ebc3f63f95e.JPG)

Hi all,
We are updating the table (attached) with the information you are sharing with us about sensors/gliders you are using/have used. 
Thank you very much for the inputs!!
Let me know if you have any comments/idea of how to improve it or if this format is useful ;) 
Patry

**Comment by charlotte-aj-williams** [comment]
For the NOC slocum (G3) gliders we have the AA4831 optode with normal foil, situated at the back of the glider! 
Thanks,
Charlotte

**Reply by gkrahmann** [reply]
At GEOMAR we still have a number of G1 Slocums with connector-potted Aanderaa 3830. At least that is the type number we have written down. All these old gliders originally had 3830 which for the deep gliders all drowned over the plugs.
Gerd

**Reply by soerenthomsen** [reply]
hi @charlotte-aj-williams and @gkrahmann ! nice to have you here! Maybe you want to take a few minutes to also introduce yourself here #47

**Comment by kimmotikka** [comment]
In FMI we have Aanderaa 4831 with a normal foil at the back of Slocum G2. Originally #404, but in the maintenance, it was replaced with #480, without notification.

**Comment by soerenthomsen** [comment]
@hahn-johannes you might want to have a look here :)

**Comment by soerenthomsen** [comment]
As many of you know already due to your contribution, we have been working on writing a Standard Operating Procedure (SOP) for How to observe Oxygen with gliders during the last months. 

We will move this document online in the next weeks here 
https://github.com/OceanGlidersCommunity/Oxygen_SOP
to allow the global glider community to feedback etc. 

Please join the https://github.com/OceanGlidersCommunity to be part of this :)
